onadata/libs/serializers/password_reset_serializer.py

```diff
@@ -12,10 +12,25 @@
 from rest_framework import serializers
 
 
+class CustomPasswordResetTokenGenerator(PasswordResetTokenGenerator):
+    """Custom Password Token Generator Class."""
+    def _make_hash_value(self, user, timestamp):
+        # Include user email alongside user password to the generated token
+        # as the user state object that might change after a password reset
+        # to produce a token that invalidated.
+        login_timestamp = '' if user.last_login is None\
+            else user.last_login.replace(microsecond=0, tzinfo=None)
+        return str(user.pk) + user.password + user.email +\
+            str(login_timestamp) + str(timestamp)
+
+
+custom_default_token_generator = CustomPasswordResetTokenGenerator()
```

Comment:
Let's just make it default_token_generator. i.e.:
```python
default_token_generator = CustomPasswordResetTokenGenerator()
```

Reply:
This is done.